### PR TITLE
fix(UIShell): reduce padding on header name to fix action button (#11687)

### LINKED ISSUE #11687: [Bug]: UI Shell menu button became squashed on `sm` breakpoint
### Package

@carbon/react

### Browser

Chrome

### Package version

1.6.0

### React version

1.6.0

### Description

Menu button became squashed at `sm` breakpoint (320px). Expecting the menu button to stay square: 

image

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?args=&id=components-ui-shell--side-nav-rail-w-header&viewMode=story

### Steps to reproduce

NA

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by asamaiya00**
UIShell: SideNav Rail w/header - reduce padding on header name to fix the width of menu action button

by: Animesh Samaiya (asamaiya00@gmail.com)

Closes #11687 

Menu button was becoming squashed somewhere between sm breakpoint (320px) and md breakpoint (672px). Fixed it by removing extra padding when md breakpoint is active. Also set the minimum width of the menu button, so that it doesn't squash further.

#### Changelog

**Changed**

packages/styles/scss/components/ui-shell/header/_header.scss

#### Testing / Reviewing

Current: 
image

Changed:
image

Please review and let me know of any changes. 

@tw15egan

### CODE REVIEW

File: packages/styles/scss/components/ui-shell/header/_header.scss [hunk 1]
```diff
@@ -42,6 +42,9 @@
     border: rem(1px) solid transparent;
     transition: background-color $duration-fast-02,
       border-color $duration-fast-02;
+    @include breakpoint-down('md') {
+      min-width: mini-units(6);
```
Comment (tw15egan) [inline packages/styles/scss/components/ui-shell/header/_header.scss]: Could we use the spacing tokens here? 

https://carbondesignsystem.com/guidelines/spacing/overview/#spacing-scale

File: packages/styles/scss/components/ui-shell/header/_header.scss [hunk 2]
```diff
@@ -137,6 +140,9 @@
     text-decoration: none;
     transition: border-color $duration-fast-02;
     user-select: none;
+    @include breakpoint-down('md') {
+      padding: 0 mini-units(2);
```
Comment (tw15egan) [inline packages/styles/scss/components/ui-shell/header/_header.scss]: Could we use the spacing tokens here? 

https://carbondesignsystem.com/guidelines/spacing/overview/#spacing-scale

**Review by tw15egan — COMMENTED** [review]
Just a couple comments

File: packages/styles/scss/components/ui-shell/header/_header.scss (hunk 1 above)
Comment (asamaiya00) [reply packages/styles/scss/components/ui-shell/header/_header.scss]: Let me try that

Comment (asamaiya00) [reply packages/styles/scss/components/ui-shell/header/_header.scss]: Yup, done @tw15egan

**Review by tw15egan — APPROVED** [review]

**Review by aledavila — APPROVED** [review]

**Pull request merged**